### DISCUSSION TITLE
Reading the code: How are resource removed (unbinded) from cluster

### DISCUSSION THREAD

**Discussion by icy** [karmada-io/karmada#4374] [General]
I've tried to read our `defaultScheduler` code and to learn how/when a cluster is not listed/matched against any propagation policy, the resources on that cluster are removed. I want to customize the scheduler to prevent that situation from happening. I haven't found the code that is in charge of this removal.  I've read at `dynamicScaleDown` but it doesn't sound what does the real job.

Would any one give me a hint?

Thanks

**Comment by XiShanYongYe-Chang** [comment]
For karmada-scheduler, each scheduling will [select the suitable clusters](https://github.com/karmada-io/karmada/blob/1cafe7fb8ff06f8fe830e4b7b1091b6e70ca5c35/pkg/scheduler/core/generic_scheduler.go#L77) and [allocate the workload](https://github.com/karmada-io/karmada/blob/1cafe7fb8ff06f8fe830e4b7b1091b6e70ca5c35/pkg/scheduler/core/generic_scheduler.go#L103) replicas. If you want to remove a cluster from the resource scheduling result, you can start from these two perspectives.
